src/listener/handlers/document-update-handler.ts

```diff
@@ -202,6 +203,30 @@ class DocumentUpdateHandler implements IHandler {
         const documentUrl = this.ingestClient.selfLinkForResource(document);
         return this.ingestClient.transitionDocumentState(documentUrl, "VALIDATING");
     }
+
+    validateFileFormat(contentValidationReport: ValidationReport, doc: any): Promise<ValidationReport> {
+        return new Promise((resolve, reject) => {
+            const fileName = doc['fileName'];
+            const fileFormat = doc['content']['file_core']['format'];
+
+            const fileExt = fileName.substring(fileName.indexOf('.') + 1);
+
+            if (fileFormat != fileExt && fileFormat && fileExt) {
+                const message = `The file extension, ${fileExt}, of the file with filename, "${fileName}", doesn't match the file format, "${fileFormat}", in the metadata.`;
+                const err = new ErrorReport(ErrorType.MetadataError, message, message);
+
+                contentValidationReport.validationState = "INVALID";
+
+                if (!contentValidationReport.validationErrors || contentValidationReport.validationErrors.length == 0) {
```

Comment:
This block is duplicated in the `checkEligibleForFileValidation` function. Consider creating a utility function